cmd/bancheck/config/config_test.go

```diff
@@ -0,0 +1,214 @@
+		files  map[string]string // fake workspace files
+		config *Config           // the expected config
+	}{
+		{
+			desc: "file with empty definitions",
+			files: map[string]string{
+				"file.json": `
+				{}
+				`,
+			},
+			config: &Config{
+				Imports:   []BannedImport{},
+				Functions: []BannedFunction{},
+			},
+		},
+		{
+			desc: "file with unknown field",
+			files: map[string]string{
+				"file.json": `
+				{
+					"unknown": 1
+				}
+				`,
+			},
+			config: &Config{
+				Imports:   []BannedImport{},
+				Functions: []BannedFunction{},
+			},
+		},
+		{
+			desc: "file with banned import",
+			files: map[string]string{
+				"file.json": `
+				{
+					"imports": [{
+						"name": "legacyconversions",
+						"msg": "Sample message",
+						"exemptions": [{
+							"justification": "My justification",
+							"allowedDir": "subdirs/vetted/..."
+						}]
+					}]
+				}
+				`,
+			},
+			config: &Config{
+				Imports: []BannedImport{
+					{
+						Name: "legacyconversions",
+						Msg:  "Sample message",
+						Exemptions: []Exemption{
+							{
+								Justification: "My justification",
+								AllowedDir:    "subdirs/vetted/...",
+							},
+						},
+					},
+				},
+				Functions: []BannedFunction{},
+			},
+		},
+		{
+			desc: "file with banned function",
+			files: map[string]string{
+				"file.json": `
+				{
+					"functions": [{
+						"name": "safehttp.NewServeMuxConfig",
+						"msg": "Sample message",
+						"exemptions": [{
+							"justification": "My justification",
+							"allowedDir": "subdirs/vetted/..."
+						}]
+					}]
+				}
+				`,
+			},
+			config: &Config{
+				Imports: []BannedImport{},
+				Functions: []BannedFunction{
+					{
+						Name: "safehttp.NewServeMuxConfig",
+						Msg:  "Sample message",
+						Exemptions: []Exemption{
+							{
+								Justification: "My justification",
+								AllowedDir:    "subdirs/vetted/...",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "multiple files",
+			files: map[string]string{
+				"file1.json": `
+				{
+					"functions": [{
+						"name": "function1"
+					}],
+					"imports": [{
+						"name": "import1"
+					}]
+				}
+				`,
+				"file2.json": `
+				{
+					"functions": [{
+						"name": "function2"
+					}],
+					"imports": [{
+						"name": "import2"
+					}]
+				}
+				`,
+			},
+			config: &Config{
+				Imports: []BannedImport{
+					{Name: "import1"},
+					{Name: "import2"},
+				},
+				Functions: []BannedFunction{
+					{Name: "function1"},
+					{Name: "function2"},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			dir, cleanup, err := analysistest.WriteFiles(test.files)
+			if err != nil {
+				t.Fatalf("Test %s: WriteFiles() returned err: %v", test.desc, err)
+			}
+			defer cleanup()
+			files := make([]string, 0)
+			for f := range test.files {
+				path := filepath.Join(dir, "src", f)
+				files = append(files, path)
+			}
+
+			config, error := Read(files)
```

Comment:
`err` is idiomatic for error. Also, you are aliasing the `error` type here.

Reply:
changed